src/csharp/Microsoft.Spark.E2ETest/UdfTests/UdfSimpleTypesTests.cs

```diff
@@ -166,5 +167,29 @@ public void TestUdfWithReturnAsTimestampType()
                 }
             }
         }
+
+        /// <summary>
+        /// UDFs defined in separate threads to validate ThreadStatic bug fix.
+        /// </summary>
+        [Fact]
+        public void TestUdfWithMultipleThreads()
```

Comment:
@imback82 Not sure if this is what you meant, or if this is a valid test. I ran this with the previous `ThreadStatic` change and it also ran successfully.

Reply:
Sorry, VStudio icon error, the message says the Test run was aborted after some seconds. Wondering how to write it so it fails.

Reply:
Can you try joining the threads?

Reply:
Joining the threads like so, still test run aborted in ThreadStatic case.

Reply:
I didn't understand the last comment. Can you debug (put breakpoint, etc.) and see what's happening?

Reply:
Debugging showed it throws the null reference exception at the UDF definition, but that exception is not thrown when running the test, or trying to catch it in a try/catch block.

Reply:
One way: you can update the delegate to catch the exception and propagate the info back to the main thread.